### PR TITLE
Feature/il 235/resetpass

### PR DESCRIPTION
**Pull request by iakivpekarskyi**
- added changing password as a guest
- added changing password as a registered user
- added password confirm validation

### CODE REVIEW

File: src/components/Auth/Modal/AuthModalLogin/AuthModal.tsx [hunk 1]
```diff
@@ -38,6 +38,10 @@ function AuthModal({ onCloseModal }: Readonly<LoginModalProps>) {
     )
   }
 
+  const handleForgotPasswordClick = () => {
+    onCloseModal?.()
```
Comment (DmitriyUshkvarok) [inline src/components/Auth/Modal/AuthModalLogin/AuthModal.tsx:42]: when you open the authorization modal window, reload the page and click on the forgot password link, the page changes and the modal window remains open.
когда во время авторизации обновляю страницу и перехожу после на форгот ссылка меняется а модалка открыта

File: src/services/authService.ts
Comment (DmitriyUshkvarok) [inline src/services/authService.ts:1]: move user endpoints to user services
authserv

File: src/validation/passwordSchema.ts
Comment (DmitriyUshkvarok) [inline src/validation/passwordSchema.ts:1]: on the client side make a hint in validation in what format to enter the password as it is done in registration confirmation
подсказку сделай в каом формате должен ввести пароль пользователь

**Review by DmitriyUshkvarok — CHANGES_REQUESTED** [review]
1)add loaders for the forgot and reset pages
2)no validation for forgot password's email inpute
3)when reloading the page the Email is on the way component is reset!
4)when the page is refreshed, Password has been changed again.
5)when the user is authorized, when the page is refreshed, the form that refers to the unauthorized user Reset your password flickers.
добавь лоадеры для страниц
отправляется запрос если поле пустое нет валидации на клиенте не одцентрован лоадер в кнопке кнопка меняет размер при загрузке
при обновлении страницы снова перебрасывает на форму
при перезагрузке мигает форма когда авторизован
при перезагрузке страницы сбрасывается стейт

File: src/services/authService.ts
Comment (iakivpekarskyi) [reply src/services/authService.ts:1]: fixed

File: src/validation/passwordSchema.ts
Comment (iakivpekarskyi) [reply src/validation/passwordSchema.ts:1]: fixed

File: src/components/Auth/Modal/AuthModalLogin/AuthModal.tsx (hunk 1 above)
Comment (iakivpekarskyi) [reply src/components/Auth/Modal/AuthModalLogin/AuthModal.tsx:42]: fixed

File: src/services/userService.ts [hunk 2]
```diff
@@ -37,3 +44,60 @@ export async function getAvatar() {
 
   return response.data
 }
+
+// Function for forgot password
+export async function apiForgotPassword(
+  email: ForgotPasswordCredentials,
+): Promise<SuccessResponse> {
+  const response: AxiosResponse<SuccessResponse> = await api.post(
+    '/auth/password/forgot',
```
Comment (DmitriyUshkvarok) [inline src/services/userService.ts:53]: move authorization-related endpoints to authorization services

File: src/components/Auth/Modal/AuthModalLogin/AuthModal.tsx [hunk 3]
```diff
@@ -38,6 +43,11 @@ function AuthModal({ onCloseModal }: Readonly<LoginModalProps>) {
     )
   }
 
+  const handleForgotPasswordClick = () => {
```
Comment (DmitriyUshkvarok) [inline src/components/Auth/Modal/AuthModalLogin/AuthModal.tsx:46]: I think the function handleForgotPasswordClick is unnecessary.
the resetOpenModal() function can be called in the onClick event of the Link component.

File: src/components/Auth/ResetPassword/AuthResetPassForm.tsx [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+import { apiAuthChangePassword, apiAuthInitPasswordChange } from '@/services/userService'
+import { yupResolver } from '@hookform/resolvers/yup'
+import { authChangePassSchema } from '@/validation/changePassSchema'
+import { IChangeAuthValues } from '@/types/ChangePassword'
+
+export default function AuthResetPassForm() {
+  const [loading, setLoading] = useState(false)
+  const { errorMessage, handleError } = useErrorHandler()
+  const { handleSubmit, register, reset, getValues } = useForm()
+  const [resetSuccessful, setResetSuccessful] = useState(false)
```
Comment (DmitriyUshkvarok) [inline src/components/Auth/ResetPassword/AuthResetPassForm.tsx:20]: I think this state should also be in the session repository.

File: src/components/Auth/ResetPassword/ForgotPassForm.tsx [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+import { useErrorHandler } from '@/services/apiError/apiError'
+import { forgotPassSchema } from '@/validation/forgotPassSchema'
+import { useLocalSessionStore } from '@/store/useLocalSessionStore'
+
+
+export default function ForgotPassForm() {
+  const [loading, setLoading] = useState(false)
+  const { errorMessage, handleError } = useErrorHandler()
+  const router = useRouter()
+  const setStoredEmailSent = useLocalSessionStore((state) => state.setStoredEmailSent)
```
Comment (DmitriyUshkvarok) [inline src/components/Auth/ResetPassword/ForgotPassForm.tsx:21]: to reduce the code, perhaps we need to destructurize the states and functions from the storage?

File: src/components/Auth/ResetPassword/ForgotPassForm.tsx [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+
+
+  const onSubmit = async () => {
+    const { email } = getValues() as { email: string }
+
+    try {
+      setLoading(true)
+      await apiForgotPassword({ email })
+      setStoredEmailSent(true)
+      localStorage.setItem('emailForReset', email)
```
Comment (DmitriyUshkvarok) [inline src/components/Auth/ResetPassword/ForgotPassForm.tsx:43]: I think email should be saved to a shared local session for more structured code

File: src/components/Auth/ResetPassword/ForgotPassForm.tsx [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+              id="email"
+              register={register}
+              name="email"
+              label="Enter your email"
+              type="email"
+              placeholder="Enter your email address"
+              className="mb-5"
+              error={errors.email}
+            />
+            <Button type="submit" className="px-6">
```
Comment (DmitriyUshkvarok) [inline src/components/Auth/ResetPassword/ForgotPassForm.tsx:101]: button resizes on server load
fix this behavior

File: src/components/Auth/ResetPassword/GuestResetPassForm.tsx [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+  })
+  const [resetSuccessful, setResetSuccessful] = useState(false)
+  const router = useRouter()
+
+  const handleButtonClick = () => {
+    router.push('/')
+  }
+
+  const onSubmit: SubmitHandler<FieldValues> = async (values) => {
+    const storedEmail = localStorage.getItem('emailForReset') ?? ''
```
Comment (DmitriyUshkvarok) [inline src/components/Auth/ResetPassword/GuestResetPassForm.tsx:40]: when the e-mail is saved in the session store, call the state from the store.

File: src/components/Auth/ResetPassword/GuestResetPassForm.tsx [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+              id="confirmPassword"
+              register={register}
+              name="confirmPassword"
+              label="Confirm new password"
+              type="password"
+              placeholder="Enter your new password again"
+              className="mb-5"
+              error={errors.confirmPassword}
+            />
+            <Button type="submit" className="my-6 px-6">
```
Comment (DmitriyUshkvarok) [inline src/components/Auth/ResetPassword/GuestResetPassForm.tsx:121]: button changes size when uploaded to the server

**Review by DmitriyUshkvarok — CHANGES_REQUESTED** [review]
if the password is successfully changed, the emailChanged state remains true. done

**Review by Sunagatov — APPROVED** [review]
Approved

**Pull request merged**